src/status_im/contexts/wallet/add_account/create_account/events.cljs

```diff
@@ -43,33 +51,45 @@
   {:fx [[:multiaccount/validate-mnemonic
          [seed-phrase
           (fn [mnemonic key-uid]
-            (rf/dispatch [:wallet/seed-phrase-validated
-                          mnemonic key-uid]))
+            (rf/dispatch [:wallet/seed-phrase-validated mnemonic key-uid]))
           on-error]]]})
 
 (rf/reg-event-fx :wallet/seed-phrase-entered seed-phrase-entered)
 
-(defn new-keypair-created
-  [{:keys [db]} [{:keys [new-keypair]}]]
-  {:db (assoc-in db [:wallet :ui :create-account :new-keypair] new-keypair)
-   :fx [[:dispatch [:navigate-back-to :screen/wallet.create-account]]]})
-
-(rf/reg-event-fx :wallet/new-keypair-created new-keypair-created)
-
-(defn new-keypair-continue
+(defn store-account-generated
+  [{:keys [db]} [{:keys [new-account-data keypair-name]}]]
+  (let [new-account (update new-account-data :mnemonic security/mask-data)]
+    {:db (-> db
+             (update-in [:wallet :ui :create-account :new-keypair]
+                        assoc
+                        :new-account-data new-account
+                        :keypair-name     keypair-name)
+             (update-in [:wallet :ui :create-account :new-keypair]
+                        dissoc
+                        :seed-phrase
+                        :random-phrase))
+     :fx [[:dispatch [:navigate-back-to :screen/wallet.create-account]]]}))
+
+(rf/reg-event-fx :wallet/store-account-generated store-account-generated)
+
+(defn generate-account-for-keypair
   [{:keys [db]} [{:keys [keypair-name]}]]
-  (let [seed-phrase (get-in db [:wallet :ui :create-account :seed-phrase])]
+  (let [seed-phrase (-> db :wallet :ui :create-account :new-keypair :seed-phrase)]
     {:fx [[:effects.wallet/create-account-from-mnemonic
-           {:seed-phrase  (security/safe-unmask-data seed-phrase)
-            :keypair-name keypair-name}]]}))
+           {:mnemonic-phrase (security/safe-unmask-data seed-phrase)
+            :paths           [constants/wallet-initial-derivation-path]
+            :on-success      (fn [new-account-data]
+                               (rf/dispatch [:wallet/store-account-generated
+                                             {:new-account-data new-account-data
+                                              :keypair-name     keypair-name}]))}]]}))
 
-(rf/reg-event-fx :wallet/new-keypair-continue new-keypair-continue)
+(rf/reg-event-fx :wallet/generate-account-for-keypair generate-account-for-keypair)
```

Comment:
Changed the event name:
`:wallet/new-keypair-continue` -> `:wallet/generate-account-for-keypair`.

Regarding to the event, it doesn't create the keypair, just generates the data for it and the account, they will be created later.